### ISSUE TITLE
Section 1 review notes

### ISSUE THREAD

**Issue by harding** [kallerosenbaum/btcphilosophy#12]
*Seems like a nifty resource!  Thanks for your work!  Here are some quick notes on section 1.  I'll try to review the other sections as I get a chance.  Feel free to ignore and just close this issue.*

# 1 Decentralization

## 1.1 Miner decentralization

This section might benefit from an examination of the non-decentralized systems that preceded Bitcoin (and followed it) and how their lack of decentralization creates problems.  E.g.

- **Centralized:** perhaps look at how David Chaum's theoretically privacy-preserving ecash solution (blinded signatures) resulted in a single-point-of-failure (SPOF) where the centralized bank had the ability to censor transactions by refusing to allow withdrawals or respending the source of the funds was identified.  In addition, the bank had the ability to mint tokens in excess of deposits or spend/lose deposited money without reporting it, silently becoming fractional reserve.

- **Distributed:** perhaps look at Hal Finney's RPOW in which a federation of Hardware Security Modules (HSMs) would sign off on transfers (transactions), permitting a threshold signing scheme that eliminated the risk of a SPOF but also requiring everyone agree on the members of the federation.  Due to the inconvenience of adding or removing members of the federation, they would be expected to become an essentially closed set.  Today, the Liquid sidechain uses this model.

- **Decentralized:** Bitcoin's advantage over the distributed federation model is that its miners exist in an open set.  Nobody needs to ask permission to become a miner, and any existing miner who stops performing essential work is automatically removed from the set.

To recover a centralized system or a distributed system from compromise, participants in the system need to appoint new coordinators.  To recovery Bitcoin from a compromise of a majority of its miners, you might not need anything besides time---the compromised miners may eventually find themselves replaced by non-compromised miners.

A good summary of some of these issues can be found in the Introduction section of https://www.blockstream.com/sidechains.pdf

## 1.2 Full node decentralization

Perhaps this could point out what full nodes are best at, which is enforcing every rule of the system that can be enforced unequivocally---that is, everything in the system except transaction ordering.

The fundamental laws of physics (specifically the maximum speed of light) force upon us equivocation about the order of transactions.  If Alice is in Europe, she's probably going to see transactions from other users in Europe faster than she sees transactions from users in North America; but if Bob is in North America, the situation is reversed for him.  This can naturally prevent Alice and Bob from independently both determining the same order of transactions, so preventing the double spending problem requires them to be dependent on some other system (mining/PoW in Bitcoin).

But every other rule in Bitcoin has been chosen to be something unequivocal---something that Alice and Bob can each independently verify without asking anyone else.  For example, the rules for signatures that prevent anyone from spending another person's bitcoins, or the rules limiting the distribution of all agreed-upon 21 million bitcoins---both of these are something that Alice and Bob can both easily verify themselves using software on their computer.  They don't need to communicate to do this, and they don't need to vote---they can simply verify whether a transaction follows all the rules and accept that transaction, or they can see that a transaction breaks the rules and reject it.

The decentralization that full nodes offer is the decentralization of being able to verify the money someone sent you is exactly the type of money you wanted to receive.  That it's not impure or corrupted or counterfeit in any way.

I wrote it, but still my favorite description of this is https://web.archive.org/web/20150929204455/https://bitcoin.org/en/bitcoin-core/features/validation

## 1.3

> It's a thing to be studied, not debate.

Lol, good luck with getting people on the internet not to debate something.  :-D

**Comment by kallerosenbaum** [comment]
Thanks Dave for your time and comments. We'll probably incorporate some of it. Keeping this open until then.

I really like your writeup on validation, I wasn't aware of that piece.

**Comment by kallerosenbaum** [comment]
@stickies-v 

Thank you for your comments!  Much appreciated!

> I think reducing it to just censorship ignores the immutability design goal, which imo is not related to censorship?

I think I disagree. Changing the ledger after the fact is censorship to me. To replace a transaction, you first have to remove the old transaction, which I think is censorship.

> I think it would be helpful to highlight why mining introduces censorship opportunities

I'm not sure what you mean here. Could you expand a bit on this?

>  I think this part is better suited in Full node decentralization? Also, I don't really think that's a censorship action.

What I'm trying to say is that the ability to censor opens up the opportunity to try to change arbitrary consensus rules. I think this should be made more clear, thanks.

> nit: repetition,

Good point!

> A decentralized network arises when multiple people agree on the same set of rules, but there is no single party in charge of creating or changing them. Anyone is free to set the rules they like, and their emergent network will consist of people that agree to the same rules. In monetary networks, there is an especially strong incentive to maintain consensus on the set of rules, but eventually it is up to each individual user.

I think I cover this in the "Full node decentralization" and "Grokking Decentralization" sections.

**Comment by kallerosenbaum** [comment]
@harding I think I've addressed your two main comments:

* Though I think your comment on "Full node decentralization" was better addressed in the first part of the [Trustlessness chapter](https://rosenbaum.se/btcphil/#trustlessness).
* Your idea on [miner decentralization](https://rosenbaum.se/btcphil/#_miner_decentralization) was addressed in a somewhat compact form.
* I didn't mention RPOW (though I wanted to) because I couldn't find information on what actually happened to the project. See https://github.com/kallerosenbaum/btcphilosophy/issues/12#issuecomment-1090193383

@stickies-v I fixed the things I thought relevant at the moment. If you disagree, I'm interested in further improvements. 

Thank you both for your input!
